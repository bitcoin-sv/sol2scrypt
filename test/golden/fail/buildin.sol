// SPDX-License-Identifier: MIT
pragma solidity ^0.8.10;

contract C {

    address private owner;

    constructor() {
        owner =  msg.sender;
    }

    function close() external { 

        bytes32 sha = keccak256(abi.encodePacked("aaa"));

        assert(msg.sender == owner);
        selfdestruct(payable(owner)); 
    }

    function recoverSignerFromSignature(uint8 v, bytes32 r, bytes32 s, bytes32 hash) external {
        address signer = ecrecover(hash, v, r, s);
        require(signer != address(0), "ECDSA: invalid signature");
    }

    function callAddMod() public pure returns(uint){
      return addmod(4, 5, 3);
    }
    
    function callMulMod() public pure returns(uint){
        return mulmod(4, 5, 3);
    }
}