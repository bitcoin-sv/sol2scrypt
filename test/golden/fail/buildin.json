[
    {
        "src": "test/golden/fail/buildin.sol:14:23:14:57",
        "type": "Error",
        "message": "unsupported function call: `keccak256`"
    },
    {
        "src": "test/golden/fail/buildin.sol:17:9:17:37",
        "type": "Error",
        "message": "unsupported function call: `selfdestruct`"
    },
    {
        "src": "test/golden/fail/buildin.sol:21:26:21:50",
        "type": "Error",
        "message": "unsupported function call: `ecrecover`"
    },
    {
        "src": "test/golden/fail/buildin.sol:26:14:26:29",
        "type": "Error",
        "message": "unsupported function call: `addmod`"
    },
    {
        "src": "test/golden/fail/buildin.sol:30:16:30:31",
        "type": "Error",
        "message": "unsupported function call: `mulmod`"
    }
]